internal/pkg/certificates/certificates.go

```diff
@@ -0,0 +1,144 @@
+package certificates
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/kubewarden/kubewarden-controller/internal/pkg/constants"
+	controllerCrypto "github.com/kubewarden/kubewarden-controller/internal/pkg/crypto"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	logger = ctrl.Log.WithName("certificates")
+)
+
+// This function will create and store in secrets, when necessary, the root CA and the certificate used in the
+// controller. If the certificate is create, the process exit to force kubernetes to redeploy the pod with the
+// latest version of the certificates.
+func RenewRootCAIfNedded(ctx context.Context, mgr ctrl.Manager, deploymentsNamespace string, controllerWebhookServiceName string) error {
+	logger.Info("Start to verify certificates")
+	k8s := mgr.GetClient()
+	rootCASecret := &corev1.Secret{}
+	err := k8s.Get(ctx, client.ObjectKey{Namespace: deploymentsNamespace, Name: constants.RootCASecretName}, rootCASecret)
+	if apierrors.IsNotFound(err) {
+		logger.Info("unable to find root CA secret.")
```

Comment:
I would move all this inner block to a dedicated function, like `createRootCA` that would generate the root CA and write that into Kubernetes.

Out of memory, I think we used to have this function

Reply:
I think you are talking about the function inside the `internal/pkg/admission` and `internal/pkg/admissionregistration`. I'm aware of that. Actually, the `internal/pkg/certificates` package is a kind of copy of some of those functions. However, I have not merge all of the code yet. Just to avoid having a huge PR as before. I'll merge the code  when I start to change the reconciliation loop for policy server and policies. For now, I'm reorganizing the necessary  functions on demand.